presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/PlanMatcher.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.assertions;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.metadata.Metadata;
+import com.facebook.presto.sql.planner.plan.PlanNode;
+import com.google.common.collect.ImmutableList;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.Collections.nCopies;
+import static java.util.Objects.requireNonNull;
+
+public final class PlanMatcher
+{
+    private final List<PlanMatcher> sources;
+    private final List<Matcher> matchers = new ArrayList<>();
+    private boolean anyTree;
+
+    public PlanMatcher(List<PlanMatcher> sources)
+    {
+        requireNonNull(sources, "sources are null");
+        requireNonNull(matchers, "matchers are null");
+
+        this.sources = ImmutableList.copyOf(sources);
+    }
+
+    public List<PlanMatchingState> matches(PlanNode node, Session session, Metadata metadata, SymbolAliases symbolAliases)
```

Comment:
This method should be package-private. PlanMatchingState is not visible outside the package, anyway.